### PR TITLE
feat: add TargetFilter::level_for_not

### PR DESCRIPTION
**Pull request by andylokandy**

### CODE REVIEW

File: src/filter/target.rs [hunk 1]
```diff
@@ -21,25 +21,37 @@ use crate::filter::FilterResult;
...
         TargetFilter {
             target: target.into(),
             level,
+            not: false,
+        }
+    }
+
+    /// The filter will be applied only if the target **does not have** a prefix that matches the
+    /// target of the log record,
+    pub fn level_for_not(target: impl Into<Cow<'static, str>>, level: log::LevelFilter) -> Self {
```
Comment (tisonkun) [inline src/filter/target.rs:44]: Read challenging. IIRC it could be generally `neg_level_for` or `negtive_level_for`?

Comment (andylokandy) [reply src/filter/target.rs:44]: It's not negative on `level` but negative on `for`. So I'd like to put not word after `for`.

Comment (andylokandy) [reply src/filter/target.rs:44]: Maybe level_for_excluding?

Comment (andylokandy) [reply src/filter/target.rs:44]: In practice, it is like:
```rust
let dispatch = Dispatch::new()
            .filter(TargetFilter::level_for_not(
                "databend::log::structlog",
                LevelFilter::Off,
            ))
            .append(structlog_log_file);
        logger = logger.dispatch(dispatch);
```

Comment (tisonkun) [reply src/filter/target.rs:44]: I'd prefer to:

```rust
pub fn negative(mut self, negative: bool) -> Self {
  self.negative = negative;
  self
}
```

Maybe use `not`, I'm OK with it. But the pattern is to invert the matching result manually in a fluent method. What do you think?

**Review by tisonkun — APPROVED** [review]
The example looks reasonable. I'm OK with either way now. You make the final call and let's try it out.

File: src/filter/target.rs [hunk 2]
```diff
@@ -21,25 +21,37 @@ use crate::filter::FilterResult;
...
+            target: target.into(),
+            level,
+            not: true,
         }
     }
 
     pub(crate) fn filter(&self, metadata: &Metadata) -> FilterResult {
-        if metadata.target().starts_with(self.target.as_ref()) {
+        let matched = metadata.target().starts_with(self.target.as_ref());
+        if (matched && !self.not) || (!matched && self.not) {
```
Comment (tisonkun) [inline src/filter/target.rs:54]: Bikeshedding - this should be logically `matched ^ self.not` 🤣

Comment (andylokandy) [reply src/filter/target.rs:54]: Yes. I did intended. It help my brain to be sane.

**Pull request merged**